### PR TITLE
Remove custom base64 encoding

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

Our functions were strange, doesn't correspond to their rfc definition (websafe writes padding but it was unused).
Also it's more code to maintain.
So I think better to remove it and use library alternative.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/1359
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -148,3 +148,6 @@ js/node/**/*.map
 js/node/**/*.map.gz
 js/apps/**/*.map
 js/apps/**/*.map.gz
+
+kek/
+test2/
```
Comment (jsteemann) [inline .gitignore:153]: :rocket:

File: lib/V8/v8-utils.cpp [hunk 2]
```diff
@@ -379,9 +381,10 @@ static void JS_Base64Decode(v8::FunctionCallbackInfo<v8::Value> const& args) {
 
   try {
     std::string const value = TRI_ObjectToString(isolate, args[0]);
-    std::string const base64 = StringUtils::decodeBase64(value);
+    std::string base64;
+    absl::Base64Unescape(value, &base64);
 
-    TRI_V8_RETURN_STD_STRING(base64);
+    TRI_V8_RETURN_STD_STRING(std::move(base64));
```
Comment (jsteemann) [inline lib/V8/v8-utils.cpp:387]: Better to not use `std::move(...)` here, given that `TRI_V8_RETURN_STD_STRING(WHAT)` is a macro and refers to its `WHAT` parameter twice. The macro always doesn't benefit from using rvalues.
```suggestion
    TRI_V8_RETURN_STD_STRING(base64);
```

File: lib/V8/v8-utils.cpp [hunk 3]
```diff
@@ -407,10 +410,10 @@ static void JS_Base64Encode(v8::FunctionCallbackInfo<v8::Value> const& args) {
   }
 
   try {
-    std::string const&& value = TRI_ObjectToString(isolate, args[0]);
-    std::string const&& base64 = StringUtils::encodeBase64(value);
+    auto value = TRI_ObjectToString(isolate, args[0]);
+    auto base64 = absl::Base64Escape(value);
 
-    TRI_V8_RETURN_STD_STRING(base64);
+    TRI_V8_RETURN_STD_STRING(std::move(base64));
```
Comment (jsteemann) [inline lib/V8/v8-utils.cpp:416]: same

File: lib/V8/v8-utils.cpp (hunk 2 above)
Comment (MBkkt) [reply lib/V8/v8-utils.cpp:387]: Understood, thanks

File: .gitignore (hunk 1 above)
Comment (MBkkt) [reply .gitignore]: Removed

File: lib/V8/v8-utils.cpp (hunk 2 above)
Comment (MBkkt) [reply lib/V8/v8-utils.cpp]: Fixed

File: lib/V8/v8-utils.cpp (hunk 3 above)
Comment (MBkkt) [reply lib/V8/v8-utils.cpp]: Fixed

File: arangod/V8Server/v8-actions.cpp [hunk 4]
```diff
@@ -840,13 +841,17 @@ static void ResponseV8ToCpp(v8::Isolate* isolate, TRI_v8_global_t const* v8g,
             // check available transformations
             if (name == "base64encode") {
               // base64-encode the result
-              out = StringUtils::encodeBase64(out);
+              std::string dest;
+              absl::Base64Unescape(out, &dest);
```
Comment (Dronplane) [inline arangod/V8Server/v8-actions.cpp:845]: Should it be Escape here?

Comment (MBkkt) [reply arangod/V8Server/v8-actions.cpp:845]: Yep

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**